### PR TITLE
Add support for calculating the inverse hyperbolic cotangent (acoth)

### LINKED ISSUE #236: RFC: add support for computing the inverse hyperbolic cotangent (acoth)
## Checklist

> Please ensure the following tasks are completed before submitting a feature request.

-   [x] Read and understood the [Code of Conduct][code-of-conduct].
-   [x] Searched for existing issues and pull requests.
-   [x] The issue name begins with `RFC:`.

## Description

> Description of the feature request.

This RFC proposes adding support for computing the inverse hyperbolic cotangent. The initial implementation is straightforward:

```javascript
return atanh( 1.0 / x );
```

Package: `@stdlib/math/base/special/acoth`
Alias: `base.acoth`

## Related Issues

> Does this feature request have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this feature request? This may include screenshots, references, sample output, and/or implementation notes.

Reference implementations:

-   [Julia](https://github.com/JuliaLang/julia/blob/dee7d77c199c4776cf4e06afb7ce920f09ad3c8c/base/special/trig.jl#L983)

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by BrunoFenzl**
Resolves #236 .

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

-   implements the `acoth` function

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #236 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/acoth/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+```javascript
+var acoth = require( '@stdlib/math/base/special/acoth' );
+```
+
+#### acoth( x )
+
+Computes the [inverse hyperbolic cotangent][inverse-hyperbolic-cotangent] of a `number` (in radians).
+
+```javascript
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/acoth/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+v = acoth( 2.0 );
+// returns ~0.5493
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/acoth/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+var acoth = require( '@stdlib/math/base/special/acoth' );
+```
+
+#### acoth( x )
+
+Computes the [inverse hyperbolic cotangent][inverse-hyperbolic-cotangent] of a `number` (in radians).
+
+```javascript
+
+v = acoth( 2.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/README.md:38]:
```suggestion
var v = acoth( 2.0 );
```

File: lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/cephes/Makefile [hunk 4]
```diff
@@ -0,0 +1,113 @@
+#/
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/cephes/Makefile]: Cephes does not have an `acoth` function. This file should be deleted.

File: lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/cephes/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,139 @@
+/**
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/cephes/benchmark.c]: Cephes does not have an `acoth` function. This file should be deleted.

File: lib/node_modules/@stdlib/math/base/special/acoth/benchmark/cpp/boost/Makefile [hunk 6]
```diff
@@ -0,0 +1,110 @@
+#/
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/benchmark/cpp/boost/Makefile]: Boost does not have an `acoth` function. This file should be deleted.

File: lib/node_modules/@stdlib/math/base/special/acoth/benchmark/cpp/boost/benchmark.cpp [hunk 7]
```diff
@@ -0,0 +1,134 @@
+/**
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/benchmark/cpp/boost/benchmark.cpp]: Boost does not have an `acoth` function. This file should be deleted.

You can verify this [here](https://github.com/boostorg/math/tree/develop/include/boost/math/special_functions).

File: lib/node_modules/@stdlib/math/base/special/acoth/docs/repl.txt [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    Returns
+    -------
+    y: number
+        Inverse hyperbolic cotangent (in radians) of a number.
+
+    Examples
+    --------
+    > var y = {{alias}}( 0.0 )
+    NaN
+    > var y = {{alias}}( 0.5 )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/docs/repl.txt]:
```suggestion
    > y = {{alias}}( 0.5 )
```

Once a variable has been declared in a REPL example, no need to redeclare.

File: lib/node_modules/@stdlib/math/base/special/acoth/docs/repl.txt [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    NaN
+    > var y = {{alias}}( 1.0 )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/docs/repl.txt]:
```suggestion
    > y = {{alias}}( 1.0 )
```

File: lib/node_modules/@stdlib/math/base/special/acoth/scripts/accuracy.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
+/**
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/scripts/accuracy.js]: This file should be removed. As no built-in implementation exists, we don't need to measure accuracy differences.

File: lib/node_modules/@stdlib/math/base/special/acoth/test/fixtures/julia/runner.jl [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+# Large positive values:
+x = linspace( 3.0, 28.0, 503 );
+gen( x, "large_positive.json" );
+
+# Larger positive values:
+x = linspace( 28.0, 100.0, 503 );
+gen( x, "larger_positive.json" );
+
+# Huge positive values:
+x = linspace( 1e300, 1e308, 1003 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/test/fixtures/julia/runner.jl]:
```suggestion
x = linspace( 1.0e300, 1.0e308, 1003 );
```

File: lib/node_modules/@stdlib/math/base/special/acoth/test/fixtures/julia/runner.jl [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+# Arguments
+
+* `x`: domain
+* `name::AbstractString`: output filename
+
+# Examples
+
+``` julia
+julia> x = linspace( -1000, 1000, 2001 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acoth/test/fixtures/julia/runner.jl]:
```suggestion
julia> x = linspace( -1000.0, 1000.0, 2001 );
```

**Review by kgryte — CHANGES_REQUESTED** [review]
@BrunoFenzl Just a couple minor changes and requests to remove extraneous files. Once these changes are made, this looks ready for merge!

File: lib/node_modules/@stdlib/math/base/special/acoth/benchmark/cpp/boost/benchmark.cpp (hunk 7 above)
Comment (BrunoFenzl) [reply lib/node_modules/@stdlib/math/base/special/acoth/benchmark/cpp/boost/benchmark.cpp]: Thanks for the link! Will be useful for the next methods

**Review by kgryte — APPROVED** [review]

**Comment by BrunoFenzl** [comment]
Yeah! That's nice :) 
I think I know where to look now when developing new features.
Thanks for your patience and the explanations!

**Comment by kgryte** [comment]
@BrunoFenzl Awesome! If you are planning on working on the other trigonometric function RFCs (e.g., `acsch`, etc), you'll want to base those efforts on [`acoth`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/acoth). I did a little bit of clean-up after merge. All-in-all, great work!

**Pull request merged**